pipeline/price_modelling/modeler_lstm_k_step.py

```diff
@@ -0,0 +1,478 @@
+import os
+import pandas as pd
+import numpy as np
+import tensorflow as tf
+from sqlalchemy import create_engine
+from datetime import datetime
+from datetime import timedelta
+from sklearn import preprocessing
+import matplotlib.pyplot as plt
+from statsmodels.nonparametric.smoothers_lowess import lowess
+import controller as ctr
+
+topicModelTable = 'TopicModel'
+model_start_date = datetime(2010, 1, 1).date()
+
+# Model parameters
```

Comment:
I remember you mentioned that you were working in best selection of hyper parameters. You used any Bayesian optimiser to find the best?

Reply:
Not for the Elasticnet, the model doesn't really have hyperparameters. The `filter_coef` is the amount of exponential smoothing, but after several trial and error, the value 1 performs significantly better (actually it is a `cumsum`). For the `bootstrapIteration`, you generally don't need to have a large sample to reduce the variance.

The selection of hyper-parameter is much more relevant for the RNN, but currently we run through a grid of values and examine the result on Tensorboard.

There are methods which are designed to specifically optimise the hyperparameters of an RNN. Please see the [learning to learn](https://arxiv.org/pdf/1606.04474.pdf), but Bayesian optimiser is also another popular option we can explore.

P.S. This section actual outlines one of the main difference between the ML/DL approach. For Elasticnet, I had to find some transformation in the data so the model predicts well, on the other hand, I didn't have to transform the data (other than normalisation) but the tuning of the model requires significant work.